### PR TITLE
Migrated db to main

### PR DESCRIPTION
**Pull request by IAmMorrow**
fixed small issue on graph tooltip
migrated db from renderer process to main process

### CODE REVIEW

**Comment by gre** [comment]
Let's try to avoid fixing graph in PR that don't relate to it ;) 

apart from this, 👏  on this, I imagine you focused on moving the logic in main.

One of the thing i'm not fan that i've noticed is the fact the debouncing is still happening in main side, that means we will send a lot of events renderer->main (for each redux update)

instead we should throttle/debounce from renderer so we don't update for each update. moreover maybe it should be one object event (`setAllTheThings({ })`) instead of 3 separated one.

It was acceptable when it was happening on the same thread, but now it will create a bottleneck if lot of events needs to happen

File: src/renderer/init.js [hunk 1]
```diff
@@ -60,17 +56,14 @@ async function init() {
...
-  db.registerTransform("app", "accounts", {
-    get: decodeAccountsModel,
-    set: encodeAccountsModel,
-  });
   const store = createStore({ dbMiddleware });
 
-  const settings = await db.getKey("app", "settings");
-  store.dispatch(fetchSettings(settings));
+  const query = querystring.parse(global.location.search);
+  const initialSettings = JSON.parse(query["?settings"]);
```
Comment (gre) [inline src/renderer/init.js:62]: I'm not fan of using location to provide the initial settings. We need a proper way to send the initial data.

File: src/main/db/defaults.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  loaded: true,
+  shareAnalytics: true,
+  sentryLogs: true,
+  lastUsedVersion: "1.18.2",
+  dismissedBanners: [],
+  accountsViewMode: "card",
+  showAccountsHelperBanner: true,
+  hideEmptyTokenAccounts: false,
+  sidebarCollapsed: false,
+};
```
Comment (MortalKastor) [inline src/main/db/defaults.js:27]: The defaults here are curious...
Why setting an arbitrary value to `lastUsedVersion` and `theme`, and `null` to `language` and `region`?

**Review by MortalKastor — COMMENTED** [review]
The new encryption implementation and migration is really neat and concise 👏
It's pretty cool you relied on the new Electron RPC stuff too 😄

Graph related changes should be moved out of the PR though.

File: src/renderer/init.js (hunk 1 above)
Comment (IAmMorrow) [reply src/renderer/init.js:62]: I could replace it with a blocking invoke during renderer init, wdyt ?

File: src/main/db/defaults.js (hunk 2 above)
Comment (IAmMorrow) [reply src/main/db/defaults.js:27]: this is a oopsie, those defaults aren't even used

File: src/renderer/init.js (hunk 1 above)
Comment (gre) [reply src/renderer/init.js:62]: i'm ok with this idea

**Pull request closed without merging**